src/Consolonia.Core/Infrastructure/AnsiConsoleOutput.cs

```diff
@@ -62,31 +74,80 @@ public void Print(PixelBufferCoordinate bufferPoint, Color background, Color for
+                {
+                    TextDecorationLocation.Strikethrough => Esc.NoStrikethrough,
+                    TextDecorationLocation.Underline => Esc.NoUnderline,
+                    _ => string.Empty
+                });
 
-            if (style == FontStyle.Italic)
-                sb.Append(Esc.Italic);
+                // Add new decoration
+                WriteText(textDecoration switch
+                {
+                    TextDecorationLocation.Underline => Esc.Underline,
+                    TextDecorationLocation.Strikethrough => Esc.Strikethrough,
+                    _ => string.Empty
+                });
+                _lastTextDecoration = textDecoration;
+            }
 
-            (object mappedBackground, object mappedForeground) =
-                consoleColorMode.Value.MapColors(background, foreground, weight);
-            sb.Append(Esc.Foreground(mappedForeground));
-            sb.Append(Esc.Background(mappedBackground));
+            if (style != _lastStyle)
+            {
+                //reset previous style
+                WriteText(_lastStyle switch
+                {
+                    FontStyle.Italic => Esc.NoItalic,
+                    _ => string.Empty
+                });
+
+                WriteText(style switch
+                {
+                    FontStyle.Italic => Esc.Italic,
+                    _ => string.Empty
+                });
+                _lastStyle = style;
+            }
+
+            if (weight != _lastWeight)
+            {
+                WriteText(weight switch
+                {
+                    FontWeight.Bold or FontWeight.SemiBold or FontWeight.ExtraBold or FontWeight.Black =>
+                        Esc.Bold,
+                    FontWeight.Thin or FontWeight.ExtraLight or FontWeight.Light => Esc.Dim,
+                    _ => Esc.Normal
+                });
+                _lastWeight = weight;
+            }
 
-            // write attributes
-            WriteText(sb.ToString());
+            if (foreground != _lastForeground || background != _lastBackground)
+            {
+                (object mappedBackground, object mappedForeground) =
+                    consoleColorMode.Value.MapColors(background, foreground, weight);
+                if (foreground != _lastForeground)
+                {
+                    WriteText(Esc.Foreground(mappedForeground));
+                    _lastForeground = foreground;
+                }
 
-            ushort textWidth = str.MeasureText();
+                if (background != _lastBackground)
+                {
+                    WriteText(Esc.Background(mappedBackground));
+                    _lastBackground = background;
+                }
+            }
 
             // move to position
             if (SupportsEmojiVariation)
             {
                 SetCaretPosition(bufferPoint);
                 WriteText(str);
+                ushort textWidth = str.MeasureText();
```

Comment:
Should not we rename Print to PrintPixel which would accept a pixel. Width would be already calculated there. Also would be no need to enumerate attributes in method parameters. Also class fields could be replaced with `_lastPixel`